scps/SCP-002.md

```diff
@@ -0,0 +1,69 @@
+## SCP-002: Datastore P2P interface
+
+This proposal is a system that allows stores to create custom p2p channels to other peers interested in a common orbitdb datastore.
+
+**Definitions**:
+* Asking peer: The peer initiating the connection
+* Responding peer: The peer responding to the initiation.
+* The network: Cloud of peers all interconnected interested in the datastore in question
+* Broadcast channel: Global pubsub channel of the corresponding datastore.
+* Application specific P2P API: Interface diagrammed by higher level code implementing `orbit-db-store`; Defined by the store type.
+
+### Hypothetical model N1
+> Note all p2p traffic is over IPFS pubsub
+
+**Flow**:
+
+1. Asking peer joins store broadcast channel. (Example: Asking peer is just starting up and gets connected to the network)
+2. Responding peer broadcasts a small message to all the other peers containing information regarding what diagrams/protocol it supports. Including Public key for encryption, if needed, and a forwarding pubsub channel ID (a channel where asking peers can send messages directly to the responding peer, derived from the peerId). The forwarding channel isn't designed to be a back and forth between a responding peer and an asking peer.  Only a temporary channel for negotiating a direct peer-to-peer connection.
```

Comment:
TODO: Research this at the IPFS level. I believe they have something like this now, or at least something experimental.

Reply:
There is some work being done for standardized data encryption as well as multikey, but it hasn't had much work done on it.
See
https://github.com/ipld/specs/issues/257
https://github.com/ipfs-inactive/docs/issues/87
https://github.com/ipfs/specs/issues/58

More clarification would be helpful